pkgs/applications/networking/ssb/patchwork/default.nix

```diff
@@ -29,16 +30,20 @@ let
     categories = "Network;";
   };
 
-in
-  symlinkJoin {
-    inherit name;
-    paths = [ binary ];
+in stdenvNoCC.mkDerivation {
+  inherit pname version;
 
-    postBuild = ''
-      mkdir -p $out/share/pixmaps/ $out/share/applications
-      cp ${appimage-contents}/ssb-patchwork.png $out/share/pixmaps
-      cp ${desktopItem}/share/applications/* $out/share/applications/
-    '';
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  buildCommand = ''
+    makeWrapper "${binary}/bin/ssb-patchwork" "$out/bin/ssb-patchwork" \
+      --prefix XDG_DATA_DIRS : "${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}" \
+      --prefix XDG_DATA_DIRS : "${gtk3}/share/gsettings-schemas/${gtk3.name}" \
```

Comment:
this is an improper use of wrapGAppsHook.
Please read https://nixos.org/nixpkgs/manual/#ssec-gnome-hooks, it exists so you don't have to do this.

It also should automatically wrap everything at `$out/bin`. If you make a symlink here for `${binary}/bin/ssb-patchwork $out/bin/ssb-patchwork` you shouldn't have to do anything.

Reply:
I would prefer an automatic solution but it doesn't work here.

Reply:
Could you explain why? If you can't you can just use the makeWrapper with the arguments wrapGAppsHook collects ` "${gappsWrapperArgs[@]}"`.